src/sqlfluff/core/rules/base.py

```diff
@@ -349,18 +349,54 @@ def _raw_slices_from_templated_slices(
 class RuleContext:
     """Class for holding the context passed to rule eval functions."""
 
-    segment: BaseSegment
-    parent_stack: Tuple[BaseSegment, ...]
-    siblings_pre: Tuple[BaseSegment, ...]
-    siblings_post: Tuple[BaseSegment, ...]
-    raw_stack: Tuple[RawSegment, ...]
-    memory: Any
+    # These don't change within a file.
     dialect: Dialect
-    path: Optional[pathlib.Path]
     fix: bool
     templated_file: Optional[TemplatedFile]
+    path: Optional[pathlib.Path]
+
+    # These change within a file.
+    segment: BaseSegment
+    parent_stack: Tuple[BaseSegment, ...] = field(default=tuple())
+    raw_segment_pre: Optional[RawSegment] = field(default=None)
+    raw_stack: Tuple[RawSegment, ...] = field(default=tuple())
+    memory: Any = field(default_factory=dict)
+    segment_idx: int = field(default=0)
+
+    @property
+    def siblings_pre(self) -> Tuple[BaseSegment, ...]:  # pragma: no cover
```

Comment:
Why no cover?

Reply:
No rules were actually using this property. When it was just a regular field, it didn't matter, but now that it's a function that's called "on demand", it's not called unless something uses it.

Reply:
I could've simply removed it, but since it's symmetrical with `siblings_post`, which _is_ used (although by only one rule itself), I thought it better to keep it.

Reply:
Updated to use ternary operator, no more "no cover", plus it's used now in this PR.